### ISSUE TITLE
[Bug] Ubuntu2404，使用默认的中文输入法打字程序会闪退

### ISSUE THREAD

**Issue by flyingcherryblossoms** [AnalyseDeCircuit/oxideterm#492]
### OxideTerm version / 版本

2.0.25

### Platform / 平台

Linux (x86_64)

### Installation source / 安装来源

GitHub Release (.dmg / .exe / .AppImage / .deb / .rpm)

### Area / 问题领域

SSH / authentication / compatibility

### Summary / 简述

安装使用的deb包

### Reproducibility / 可复现性

Always (100%) / 必现

### Steps to reproduce / 复现步骤

打开程序，新建连接，输入中文

### Expected vs actual / 预期与实际

程序直接退出

### Checklist

- [x] I tested with the latest release, can still reproduce the issue, and searched existing issues / 我已使用最新正式版测试，问题仍然存在，并已搜索过已有 Issue
- [x] This is one reproducible bug, not a usage question or feature request / 这是一个可复现的 bug，而不是使用问题或功能建议
- [x] I provided the OxideTerm version, platform, and steps to reproduce; I understand vague, incomplete, or inactive issues may be closed.
我已提供 OxideTerm 版本、平台及复现步骤；我理解描述模糊、信息不足或长期无回复的 Issue 可能会被关闭。

- [x] I removed passwords, private keys, and other secrets from this report.
我已从本报告中删除密码、私钥及其他敏感信息。

**Comment by AnalyseDeCircuit** [comment]
当前信息不足以定位 Linux 输入法崩溃，请补充：

1. 输入中文时发生崩溃的具体字段，例如连接名称、备注或用户名。
2. 执行以下命令并贴出结果：

```bash
echo "$XDG_CURRENT_DESKTOP"
echo "$XDG_SESSION_TYPE"
ibus version 2>/dev/null || fcitx5 --version
```

3. 从终端启动 OxideTerm，复现后附上完整崩溃输出：

```bash
RUST_BACKTRACE=1 /opt/oxideterm/oxideterm-native 2>&1 | tee /tmp/oxideterm-ime-crash.log
```

请在上传前删除日志中的主机名、IP、用户名和其他敏感信息。Wayland 与 X11 使用不同的输入法路径，需要上述环境和堆栈信息才能继续定位。

**Comment by flyingcherryblossoms** [comment]
1.只要切换中文在任何地方输入都会崩溃，不仅仅在新建连接的地方
2.
echo "$XDG_CURRENT_DESKTOP"
ubuntu:GNOME
echo "$XDG_SESSION_TYPE"
x11
ibus version 2>/dev/null || fcitx5 --version
IBus 1.5.29-rc2

**日志**
RUST_BACKTRACE=1 /opt/oxideterm/oxideterm-native 2>&1 | tee /tmp/oxideterm-ime-crash.log
thread 'main' (21954) panicked at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zed-xim-0.4.0-zed/src/client.rs:237:67:
Encoding Error: UnsupportedEncoding
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

**详细日志**
RUST_BACKTRACE=full /opt/oxideterm/oxideterm-native 2>&1 | tee /tmp/oxideterm-ime-crash.log

thread 'main' (22183) panicked at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zed-xim-0.4.0-zed/src/client.rs:237:67:
Encoding Error: UnsupportedEncoding
stack backtrace:
   0:     0x63a56dfba743 - <unknown>
   1:     0x63a5698d817a - <unknown>
   2:     0x63a56df7c0c6 - <unknown>
   3:     0x63a56df93ad9 - <unknown>
   4:     0x63a56df9390f - <unknown>
   5:     0x63a56df93d52 - <unknown>
   6:     0x63a56df93b98 - <unknown>
   7:     0x63a56df8d209 - <unknown>
   8:     0x63a56df6f6fd - <unknown>
   9:     0x63a5698e349c - <unknown>
  10:     0x63a5698e2582 - <unknown>
  11:     0x63a56a2ac6b1 - <unknown>
  12:     0x63a56a1172c2 - <unknown>
  13:     0x63a56a196cbc - <unknown>
  14:     0x63a56a0e0bba - <unknown>
  15:     0x63a56a2f90bd - <unknown>
  16:     0x63a56b7e9b65 - <unknown>
  17:     0x63a56c4712f3 - <unknown>
  18:     0x63a56c4626a9 - <unknown>
  19:     0x63a56df7dcd6 - <unknown>
  20:     0x63a56cced195 - <unknown>
  21:     0x7d25ae82a1ca - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  22:     0x7d25ae82a28b - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:360:3
  23:     0x63a5695f08e5 - <unknown>
  24:                0x0 - <unknown>

看起来是编码问题

**Comment by AnalyseDeCircuit** [comment]
已处理并推送至 2809aa565。根因是 X11/IBus 发送的 GB2312 Compound Text 未被原解码器支持，而 zed-xim 又对解码失败使用了强制终止，最终导致主线程崩溃。

现在已经补齐中文 Compound Text 解码，并将预编辑、提交和重置路径中的解码失败改为可恢复错误；即使输入法以后发送未知或损坏的编码，也不会再因此直接退出整个应用。

此改动会随下一次构建或发布进入可用版本，届时请在 Ubuntu 24.04、X11、IBus 环境下复验中文预编辑和提交。

**Comment by flyingcherryblossoms** [comment]
好的，经测试问题已经修复，辛苦
